tests/test_single_topology_v3.py

```diff
@@ -0,0 +1,180 @@
+    assert set(improper_idxs) == set()
+
+    # set [H] as the dummy group
+    all_idxs, _ = setup_dummy_interactions_from_ff(ff, mol, dummy_group=[12], root_anchor_atom=6, nbor_anchor_atom=5)
+    bond_idxs, angle_idxs, improper_idxs = all_idxs
+
+    assert set(bond_idxs) == set([(6, 12)])
+    assert set(angle_idxs) == set([(5, 6, 12)])
+    assert set(improper_idxs) == set()
+
+    with pytest.raises(single_topology_v3.MissingAngleError):
+        all_idxs, _ = setup_dummy_interactions_from_ff(
+            ff, mol, dummy_group=[12], root_anchor_atom=6, nbor_anchor_atom=4
+        )
+
+
+def test_find_dummy_groups_and_anchors():
+    """
+    Test that we can find the anchors and dummy groups when there's a single core anchor atom. When core bond
+    is broken, we should disable one of the angle atoms.
+    """
+    mol_a = Chem.MolFromSmiles("OCCC")
+    mol_b = Chem.MolFromSmiles("CCCF")
+    core = np.array([[1, 2], [2, 1], [3, 0]])
+
+    dgs, jks = single_topology_v3.find_dummy_groups_and_anchors(mol_a, mol_b, core[:, 0], core[:, 1])
+    assert dgs == [{3}]
+    assert jks == [(2, 1)]
+
+    # angle should swap
+    core = np.array([[1, 2], [2, 0], [3, 1]])
+
+    with pytest.warns(CoreBondChangeWarning):
+        dgs, jks = single_topology_v3.find_dummy_groups_and_anchors(mol_a, mol_b, core[:, 0], core[:, 1])
+        assert dgs == [{3}]
+        assert jks == [(2, None)]
+
+
+def testing_find_dummy_groups_and_multiple_anchors():
+    """
+    Test that we can find anchors and dummy groups with multiple anchors, we expect to find only a single
+    root anchor and neighbor core atom pair.
+    """
+    mol_a = Chem.MolFromSmiles("OCC")
+    mol_b = Chem.MolFromSmiles("O1CC1")
+    core = np.array([[1, 1], [2, 2]])
+
+    with pytest.warns(MultipleAnchorWarning):
+        dgs, jks = single_topology_v3.find_dummy_groups_and_anchors(mol_a, mol_b, core[:, 0], core[:, 1])
+        assert dgs == [{0}]
+        assert jks == [(1, 2)] or jks == [(2, 1)]
+
+    mol_a = Chem.MolFromSmiles("C(C)(C)C")
+    mol_b = Chem.MolFromSmiles("O1CCCC1")
+
+    core_a = [0, 1, 2, 3]
+    core_b = [2, 1, 4, 3]
+
+    with pytest.warns(MultipleAnchorWarning):
+        dgs, jks = single_topology_v3.find_dummy_groups_and_anchors(mol_a, mol_b, core_a, core_b)
+        assert dgs == [{0}]
+        assert jks == [(1, 2)]
+
+
+def compute_distance_matrix(x0):
+    N = x0.shape[0]
+    D = x0.shape[1]
+    xi = x0.reshape((1, N, D))
+    xj = x0.reshape((N, 1, D))
+    rij = xi - xj
+    dij = jnp.linalg.norm(rij, axis=-1)
+    return dij
+
+
+def get_max_distance(x0):
+    dij = compute_distance_matrix(x0)
+    return jnp.amax(dij)
+
+
+def test_hif2a_end_state_stability():
+    """
+    Pick some random pairs from the hif2a set and ensure that they're numerically stable at the
+    end-states under some a standard distance based atom_mapping protocol.
+    """
+    suppl = Chem.SDMolSupplier("timemachine/testsystems/data/ligands_40.sdf", removeHs=False)
```

Comment:
nit: better to use
```suggestion
    with resources.path("timemachine.testsystems.data", "ligands_40.sdf") as path_to_ligand:
        suppl = Chem.SDMolSupplier(str(path_to_ligand), removeHs=False)
```
to explicitly use the packaged test data ([example](https://github.com/mcwitt/timemachine/blob/45b973f312fedaa8fe4693b6c32bb481f8afbb40/tests/test_handlers.py#L685-L686))

Reply:
willfix inline